### PR TITLE
#5481 update to ipywidgets v7

### PR DESCRIPTION
**Pull request by michalgce**

### CODE REVIEW

**Comment by scottdraves** [comment]
let's wait until v7 appears in conda

**Comment by scottdraves** [comment]
screen shot 2017-08-23 at 10 33 57 am
EasyForm not syncing its values back to Kernel

**Comment by scottdraves** [comment]
i can safely switch on and off this branch by using
```
conda install -c conda-forge ipywidgets=7.0.0
```
and
```
conda install -c conda-forge widgetsnbextension=2.0.0 ipywidgets=6.0.0
```

File: beakerx/js/webpack.config.js [hunk 1]
```diff
@@ -142,7 +142,7 @@ module.exports = [
       modules: ['web_modules', 'node_modules'],
       extensions: ['.jsx','.js','.less','.css']
     },
-    externals: ['jupyter-js-widgets', '@jupyter-widgets/base'],
+    externals: ['@jupyter-widgets/controls', '@jupyter-widgets/base', '@jupyter-widgets/base'],
```
Comment (MariuszJurowicz) [inline beakerx/js/webpack.config.js]: `@jupyter-widgets/base` duplicated

File: kernel/base/src/main/java/com/twosigma/beakerx/easyform/EasyFormComponent.java [hunk 2]
```diff
@@ -150,4 +155,23 @@ public void close() {
...
+      return;
+    }
+    Map<String, Map<String,String>> dataMap = ((Map<String, Map<String, String>>) message.getContent().get(DATA));
+    if (dataMap != null && !dataMap.isEmpty()) {
+      Map<String, String> stateMap = dataMap.get(STATE);
+      if (stateMap != null && !stateMap.isEmpty()) {
+        Object newValue = null;
+        if (stateMap.containsKey("value")) {
+          newValue = stateMap.get("value");
+        } else if (stateMap.containsKey("index")) {
```
Comment (MariuszJurowicz) [inline kernel/base/src/main/java/com/twosigma/beakerx/easyform/EasyFormComponent.java]: Maybe this whole `if` statement could be simplified a little

File: kernel/base/src/main/java/com/twosigma/beakerx/widgets/selections/SelectionWidget.java [hunk 3]
```diff
@@ -54,4 +54,12 @@ public void setSize(Integer size) {
     sendUpdate(SIZE, size);
   }
 
+  public String getModelModuleValue(){
+    return "@jupyter-widgets/controls";
+  }
+
+  public String getViewModuleValue(){
+    return "@jupyter-widgets/controls";
```
Comment (MariuszJurowicz) [inline kernel/base/src/main/java/com/twosigma/beakerx/widgets/selections/SelectionWidget.java]: Is this change necessary while in other examples we are using `jupyter-js-widgets` ?

**Review by MariuszJurowicz — CHANGES_REQUESTED** [review]

**Review by MariuszJurowicz — APPROVED** [review]

**Pull request merged**